p2p/peer.py

```diff
@@ -620,10 +531,21 @@ def max_headers_fetch(self) -> int:
         return les.MAX_HEADERS_FETCH
 
     def handle_sub_proto_msg(self, cmd: protocol.Command, msg: protocol._DecodedMsgType) -> None:
+        cmd_type = type(cmd)
+
         if isinstance(cmd, les.Announce):
             self.head_info = cmd.as_head_info(msg)
             self.head_td = self.head_info.total_difficulty
             self.head_hash = self.head_info.block_hash
+        elif cmd_type in self.pending_requests:
+            request, future = self.pending_requests[cmd_type]
+            try:
+                request.validate_response(msg)
+            except ValidationError:
+                pass
```

Comment:
We should log this, probably as a warning even, no?

Reply:
Yes, though I've been questioning the only one request of a single type in-flight per message type restriction.  I'm going to keep it in place, but unless the spec says that it's disallowed, I'd like to support it eventually.  Sending concurrent requests to high performing peers seems like a way to speed up sync.

Reply:
Fair enough, but IIRC geth doesn't do that, so it might be worth finding out why before we consider doing it ourselves.